src/AppBundle/Controller/Admin/BlogController.php

```diff
@@ -72,7 +72,14 @@ public function newAction(Request $request)
     {
         $post = new Post();
         $post->setAuthorEmail($this->getUser()->getEmail());
-        $form = $this->createForm(new PostType(), $post);
+
+        // We need a form builder to dynamically add a submit button.
+        // See http://symfony.com/doc/current/book/forms.html#submitting-forms-with-multiple-buttons
+        $form = $this
+            ->get('form.factory')
+            ->createBuilder(new PostType(), $post)
+            ->add('createAndShow', 'submit')
+            ->getForm();
```

Comment:
@yceruto, `createForm` returns an instance of `Form` class, but we need an instance of `FormBuilder` to add a button dynamically.